[browser][wasm] Default Debug Write to StdErr

kjpou1 opened the pull request:
- COMPlus_DebugWriteToStdErr environment variable when set to `1` will write to `stderr` so it show up in the browser console log.

kjpou1 committed:
[browser][wasm] Default Debug Write to StdErr

- COMPlus_DebugWriteToStdErr environment variable when set to `1` will write to `stderr` so it show up in the browser console log.

kg commented on src/mono/wasm/runtime/driver.c:
```diff
@@ -315,6 +315,7 @@ mono_wasm_load_runtime (const char *managed_path, int enable_debugging)
 
 	//monoeg_g_setenv ("MONO_LOG_LEVEL", "debug", 0);
 	//monoeg_g_setenv ("MONO_LOG_MASK", "gc", 0);
+	monoeg_g_setenv ("COMPlus_DebugWriteToStdErr", "1", 0);
```
Maybe add a comment explaining why we're setting this so that people don't have to git blame and then look at the PR

marek-safar replied:
Why are we even setting this in release build?

kjpou1 replied:
We will need to add more to the build so that the driver build knows not to set in release.

akoeplinger replied:
It should be a no-op on Release builds already since all of the `Debug` methods are under a `System.Diagnostics.Conditional("DEBUG")]`.

vargaz replied:
The System.Diagnostics.DebugProvider.WriteCore () method is the only way to emit debug logging from inside corlib, and it requires this env var to be set to work. Note that stderr handling is still buggy somewhere, so the logs will still not show up.

marek-safar replied:
> It should be a no-op on Release builds

Will this be no-op on wasm runtime side?

akoeplinger replied:
@marek-safar it doesn't really have anything to do with the wasm runtime, any Release build of the BCL will have the `Debug.*` methods as no-op.

marek-safar replied:
What I meant is that monoeg_g_setenv will add size cost on runtime if this is the only usage which might be possible

kjpou1 replied:
There are multiple occurrences but the main case is that we also provide a public interface to `setenv` so environment variables can be set.

https://github.com/dotnet/runtime/blob/514455aafebc289916ee622608255279c7225a4f/src/mono/wasm/runtime/driver.c#L158

as well as https://github.com/dotnet/runtime/blob/514455aafebc289916ee622608255279c7225a4f/src/mono/wasm/runtime/driver.c#L158

kjpou1 replied:
> Maybe add a comment explaining why we're setting this so that people don't have to git blame and then look at the PR

Addressed in recent commit

kjpou1 committed:
Address review comment about adding a comment

kjpou1 committed:
Address review comments

Add conditional compile for DEBUG build to only set debug environment variables when /p:Configuration=Debug is specified
